File: day3/solution part 1.cpp
#include <iostream>
#include <regex>
#include <string>
#include <fstream>

int main(){

    std::ifstream file("data.txt");
    std::string content((std::istreambuf_iterator<char>(file)), std::istreambuf_iterator<char>());

    file.close();

    std::regex pattern(R"(mul\((\d+),(\d+)\))");

    std::sregex_iterator begin(content.begin(), content.end(), pattern);
    std::sregex_iterator end;

    long long int sum{};
    for (auto it = begin; it != end; ++it){
        std::smatch match = *it;
        int lhs = stoi(match[1].str());
        int rhs = stoi(match[2].str());
        sum += (lhs * rhs);
    }
    std::cout << sum;
    return 0;
}

File: day3/solution part 1.py
import re

pattern = r"mul\((\d+),(\d+)\)"             # match mul(int,int) and add them to a list

with open("data.txt", 'r') as file:
    content = file.read()

matches = re.findall(pattern, content)

sum = 0

for expression in matches:                  # convert to int and sum up the multiplication
    sum += int(expression[0]) * int(expression[1])

print(sum)

